src/mbgl/map/map_context.cpp

```diff
@@ -112,7 +112,12 @@ void MapContext::setStyleURL(const std::string& url) {
         if (res.status == Response::Successful) {
             loadStyleJSON(res.data, base);
         } else {
-            Log::Error(Event::Setup, "loading style failed: %s", res.message.c_str());
+            if (res.message.substr(res.message.length() - 3) == "404" &&
```

Comment:
Hmmm, apparently we could be done by adding an `else if (responseCode == 404) {` there.